File: Week_6/readability.py
def main():
    text = input("Text: ")      # get input text

    index = get_index(text)     # calculate index
    print_grade(index)          # print "Grade X" based on index


def get_index(text):
    words = 1       # start at 1 because it counts by spaces e.g. "Dogs. Cats."
    letters = sentences = 0

    for char in text:   # check each character in text string
        if char == " ":
            words += 1
        if char in ("!", "?", "."):
            sentences += 1
        if char.isalpha():
            letters += 1

    L = (letters / words) * 100   # (letters / words) is number of letters per 1 word
    S = (sentences / words) * 100

    return (0.0588 * L - 0.296 * S - 15.8)


def print_grade(index):     # print "Grade X" based on index
    grade = round(index)    # round it to nearest int

    if grade >= 16:
        print("Grade 16+")
        return
    elif grade < 1:
        print("Before Grade 1")
        return
    else:
        print(f"Grade {grade}")


main()

File: Week_2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

void process_text(string text) // printf after getting text input from user
{
    int words = 1, letters = 0, sentences = 0; // starts with words = 1 as last word has no space
    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] >= 65 && text[i] <= 90) || (text[i] >= 97 && text[i] <= 122))
        {
            letters++;
        }
        if (text[i] == ' ') // space denotes word.
        {
            words++;
        }
        if (text[i] == '!' || text[i] == '?' || text[i] == '.')
        {
            sentences++;
        }
    }

    float L = 100.0 * letters / words; // 100.0 converts whole thing to float
    float S = 100.0 * sentences / words; // avg no. of sentences per 100 words

    float index = 0.0588 * L - 0.296 * S - 15.8;
    int int_index = (int) round(index);

    if (int_index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (int_index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", int_index);
    }
}

int main(void)
{
    string text = get_string("Text: "); // get string
    process_text(text);
}